addons/edi/models/edi_record.py

```diff
@@ -280,16 +295,15 @@ def prepare(self, doc, vlist):
         """
 
         # Initialise statistics
-        count = 0
         _logger.info("%s preparing %s", doc.name, self._name)
 
         # Create records
         with self.statistics() as stats:
             try:
-                for record_vals in vlist:
-                    record_vals['doc_id'] = doc.id
-                    self.create(record_vals)
-                    count += 1
+                values_to_write = [
```

Comment:
It looks a little confusing the variable name values_to_write which is used in .create() :(
Maybe smth like vals can do the job